graphql-dgs-codegen-gradle/src/main/kotlin/com/netflix/graphql/dgs/codegen/gradle/ClientUtilsConventions.kt

```diff
@@ -0,0 +1,77 @@
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.graphql.dgs.codegen.gradle
+
+import org.gradle.api.Project
+import org.gradle.api.logging.Logging
+import java.util.*
+
+object ClientUtilsConventions {
+
+    private const val CLIENT_UTILS_ARTIFACT_GROUP = "com.netflix.graphql.dgs.codegen"
+    private const val CLIENT_UTILS_ARTIFACT_NAME = "graphql-dgs-codegen-client-utils"
+    private const val GRADLE_CLASSPATH_CONFIGURATION = "compileClasspath"
+
+    private val logger = Logging.getLogger(ClientUtilsConventions::class.java)
+
+    fun apply(project: Project, optionalCodeUtilsVersion: Optional<String> = Optional.empty()) {
+        clientUtilsArtifact(optionalCodeUtilsVersion).ifPresent { dependencyString ->
+            val implementationClasspath = project.configurations.getByName(GRADLE_CLASSPATH_CONFIGURATION).dependencies
+            implementationClasspath.add(project.dependencies.create(dependencyString))
+            logger.lifecycle("DGS CodeGen added [$dependencyString] to the $GRADLE_CLASSPATH_CONFIGURATION.")
+        }
+    }
+
+//    fun apply(project: Project, optionalCodeUtilsVersion: Optional<String> = Optional.empty()) {
+//        project.configurations
+//                .matching { it.name == "compileClasspath" || it.name == "implementation" }
+//                .all { configuration ->
+//                    val explicitClientUtilsDependencies = configuration.dependencies.filter(this::findClientUtils)
+//                    if (explicitClientUtilsDependencies.isEmpty()) {
+//                        clientUtilsArtifact(optionalCodeUtilsVersion).ifPresent { dependencyString ->
+//                            val implementationClasspath = project.configurations.getByName(GRADLE_CLASSPATH_CONFIGURATION).dependencies
+//                            implementationClasspath.add(project.dependencies.create(dependencyString))
+//                            logger.lifecycle("DGS CodeGen added [$dependencyString] to the $GRADLE_CLASSPATH_CONFIGURATION.")
+//                        }
+//                    }
+//                }
+//    }
+
+//    private fun findClientUtils(dependency: Dependency): Boolean {
+//        return CLIENT_UTILS_ARTIFACT_GROUP == dependency.group &&
+//                CLIENT_UTILS_ARTIFACT_NAME == dependency.name
+//    }
+
+    private val pluginProperties: Optional<Properties> = try {
+        val props = Properties()
+        props.load(this.javaClass.classLoader.getResourceAsStream("META-INF/graphql-dgs-codegen-core.properties"))
```

Comment:
what is this?

Reply:
In order to add the dependency I need to the `client-core` module to the project I need to infer its version. To do so I'm reading the _build information_, as provided by the [nebula.info plugin](https://github.com/nebula-plugins/gradle-info-plugin).